src/DataHandling/Logs/DataDogAudit.php

```diff
@@ -165,22 +271,242 @@ protected function removeColumnIfOnlyUnchanged($diffArray)
 
     /**
      * Method for getting key storing information about given on input column name with association. Can be override for customization needs.
-     * @param \DataDog\AuditBundle\Entity\AuditLog $currentVersion
-     * @return string name 
+     *
+     * @param AuditLog $currentVersion
+     *
+     * @return string name
      */
     protected function getColumnNameForAssociation(AuditLog $currentVersion)
     {
-        return $currentVersion->getTarget()->getTbl();
+        $joinTableName = $currentVersion->getTbl();
+        if (isset($this->cache['assocTable'][$joinTableName])) {
+            return $this->cache['assocTable'][$joinTableName];
+        }
+
+        foreach ($this->getEntitiesClassMetaData()->getAssociationMappings() as $assMapping) {
+            if (isset($assMapping['joinTable'])) {
+                $assJoinTable = $assMapping['joinTable']['name'];
+                $this->cache['assocTable'][$assJoinTable] = $assMapping['fieldName'];
+                if ($joinTableName === $assJoinTable) {
+                    return $assMapping['fieldName'];
+                }
+            } elseif (is_null($currentVersion->getTarget())
+                && $currentVersion->getSource()->getClass() === $assMapping['targetEntity']
+            ) { // ManyToOne mapping
+                $this->cache['assocTable'][$joinTableName] = $assMapping['fieldName'];
+
+                return $assMapping['fieldName'];
+            }
+        }
+
+        throw new \LogicException(sprintf('no association for %s found.', $joinTableName));
+    }
+
+    protected function getLabelForAssociation(Association $ass)
```

Comment:
I would rather not use this variable name in our code (see google translator) ;)

Reply:
agree 😞